src/include/network/lan/address.rb

```diff
@@ -1515,7 +1518,7 @@ def update_hostname(ipaddr, hostname)
       initial_hostname = initial_hostname(LanItems.ipaddr)
       hostname_changed = initial_hostname != hostname
 
-      return if !(ip_changed || hostname_changed || hostname.empty?)
+      return if !(ip_changed || hostname_changed) || hostname.empty?
```

Comment:
It respect the method description? I mean, if you have a previous `hostname` and you clear it, the hostname is not updated / removed from /etc/hosts?

Reply:
good point ... I've to make the condition even more strict.